### ISSUE TITLE
There is a delay in getting the frontmost app!

### ISSUE THREAD

**Issue by ayangweb** [HuakunShen/applications-rs#3]
There is a delay of at least 3s when fetching the frontmost apps! And when I fetch all apps, most of them don't get icons!

**Comment by HuakunShen** [comment]
I took 947.209µs for me.

Ran this with `cargo run --release`

```rust
use applications::{AppInfoContext, AppInfo};

fn main() {
    let mut ctx = AppInfoContext::new();
    let start_time = std::time::Instant::now();
    let frontmost_app = ctx.get_frontmost_application().unwrap();
    println!("Frontmost App: {:#?}", frontmost_app);
    println!("Time: {:?}", start_time.elapsed());
}



// Frontmost App: App {
//     name: "Code",
//     icon_path: Some(
//         "/Applications/Visual Studio Code.app/Contents/Resources/Code.icns",
//     ),
//     app_path_exe: Some(
//         "/Applications/Visual Studio Code.app/Contents/MacOS/Electron",
//     ),
//     app_desktop_path: "/Applications/Visual Studio Code.app",
// }
// Time: 947.209µs
```

What did you run?

**Comment by ayangweb** [comment]
> I took 947.209µs for me.
> 
> Ran this with `cargo run --release`
> 
> ```rust
> use applications::{AppInfoContext, AppInfo};
> 
> fn main() {
>     let mut ctx = AppInfoContext::new();
>     let start_time = std::time::Instant::now();
>     let frontmost_app = ctx.get_frontmost_application().unwrap();
>     println!("Frontmost App: {:#?}", frontmost_app);
>     println!("Time: {:?}", start_time.elapsed());
> }
> 
> 
> 
> // Frontmost App: App {
> //     name: "Code",
> //     icon_path: Some(
> //         "/Applications/Visual Studio Code.app/Contents/Resources/Code.icns",
> //     ),
> //     app_path_exe: Some(
> //         "/Applications/Visual Studio Code.app/Contents/MacOS/Electron",
> //     ),
> //     app_desktop_path: "/Applications/Visual Studio Code.app",
> // }
> // Time: 947.209µs
> ```
> 
> What did you run?

This code is really fast at getting app information, but if you switch apps at the moment of copying, it will still result in a printout of the app you've switched to, As you can see from the video, I copied google's content and as soon as I switched windows, it printed nothing but vscode information and vice versa!


https://github.com/HuakunShen/applications-rs/assets/75017711/274ce6c4-b9be-45f8-b09f-b8ba646eb0fd

**Comment by HuakunShen** [comment]
Yeah, I tried. If you switch too fast you will get the wrong app.
I don't have a solution for now. The best alternative is to get the active app in rust code when clipboard update is detected, instead of waiting for the event in JS and query the active app. This could potentially significantly reduce latency. I am not sure.

**Comment by ayangweb** [comment]
> Yeah, I tried. If you switch too fast you will get the wrong app. I don't have a solution for now. The best alternative is to get the active app in rust code when clipboard update is detected, instead of waiting for the event in JS and query the active app. This could potentially significantly reduce latency. I am not sure.

Yeah, I was thinking the same thing, and having rust do it in a timely manner might be a good note for this one

**Comment by ayangweb** [comment]
> Yeah, I tried. If you switch too fast you will get the wrong app. I don't have a solution for now. The best alternative is to get the active app in rust code when clipboard update is detected, instead of waiting for the event in JS and query the active app. This could potentially significantly reduce latency. I am not sure.

<img width="946" alt="image" src="https://github.com/HuakunShen/applications-rs/assets/75017711/dd239000-543d-4fd3-999f-ef09392dd5cc">

Hey👋, I want to try rust to get app info, but rust can't get the `onSomethingUpdate` event, is it possible to expose it? Is it possible to expose it and write an example (I'm not very familiar with rust)? Thanks!

**Comment by HuakunShen** [comment]
@ayangweb "plugin:clipboard://something-changed" is emitted, you can try to listen to this event in rust.

**Comment by ayangweb** [comment]
> @ayangweb "plugin:clipboard://something-changed" is emitted, you can try to listen to this event in rust.

<img width="885" alt="image" src="https://github.com/HuakunShen/applications-rs/assets/75017711/8f16a629-58e5-49e5-9647-df746b4ad3e7">

Is this how it is written please? It didn't trigger on my end, please help me, thanks!
